File: katas/8 kyu/opposite-number/solution.php
<!-- opposite-number -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function opposite($n) {
  return -$n;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
class OppositeTest extends TestCase {
  public function testThatOppositeWorksForExamplesProvidedInDescription() {
    $this->assertSame(opposite(1), -1);
    $this->assertSame(opposite(14), -14);
    $this->assertSame(opposite(-34), 34);
  }
}

File: katas/8 kyu/opposite-number/solution.py
# opposite-number
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def opposite(number):
  return -number
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(opposite(1),-1)
test.assert_equals(opposite(25.6), -25.6)
test.assert_equals(opposite(0), 0)
test.assert_equals(opposite(1425.2222), -1425.2222)
test.assert_equals(opposite(-3.1458), 3.1458)
test.assert_equals(opposite(-95858588225),95858588225)